### PR TITLE
Add comments (+ questions) to matrix assembly 

### PR DESCRIPTION
**Pull request by maxpaik16**
Added comments to main assemble function

Added comments to Laplacian and Mass Assembler classes

Added some comments to MatrixCache class

Added clarifying questions to MatrixCache, error computation, and Laplacian local assembly function. If there are answers to these, I can update the comments accordingly.

### CODE REVIEW

File: src/polyfem/assembler/Laplacian.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@ namespace polyfem::assembler
 		const Eigen::MatrixXd &gradj = data.vals.basis_values[data.j].grad_t_m;
 		// return ((gradi.array() * gradj.array()).rowwise().sum().array() * da.array()).colwise().sum();
 		double res = 0;
+		// why not just loop over quadrature points here?
```
Comment (teseoch) [inline src/polyfem/assembler/Laplacian.cpp:18]: it is the same, maybe you can add an assert to make it explicit

File: src/polyfem/utils/MatrixCache.hpp [hunk 2]
```diff
@@ -84,14 +85,14 @@ namespace polyfem::utils
 	private:
 		size_t size_;
 		StiffnessMatrix tmp_, mat_;
-		std::vector<Eigen::Triplet<double>> entries_;
-		std::vector<std::vector<std::pair<int, size_t>>> mapping_;
+		std::vector<Eigen::Triplet<double>> entries_; ///< contains global matrix indices and corresponding value
+		std::vector<std::vector<std::pair<int, size_t>>> mapping_; ///< maps row indices to column index/local index pairs
 		std::vector<int> inner_index_, outer_index_;
-		std::vector<double> values_;
+		std::vector<double> values_; ///< maps local index to corresponding value
```
Comment (teseoch) [inline src/polyfem/utils/MatrixCache.hpp:91]: no, this contains the values

File: src/polyfem/assembler/Assembler.cpp [hunk 3]
```diff
@@ -225,6 +231,7 @@ namespace polyfem::assembler
 										continue;
 									}
 
+									// loop over the global nodes corresponding to local element
```
[the remark this answers is not in the document]
Comment (maxpaik16) [reply src/polyfem/assembler/Assembler.cpp:234]: I see. Is it worth trying to document a bit more or should I not worry about it for now?

File: src/polyfem/assembler/Laplacian.cpp (hunk 1 above)
Comment (maxpaik16) [reply src/polyfem/assembler/Laplacian.cpp:18]: Sure, I can add one.

File: src/polyfem/state/StateSolveLinear.cpp [hunk 4]
```diff
@@ -96,6 +96,8 @@ namespace polyfem
 		solver->getInfo(stats.solver_info);
 
 		const auto error = (A * x - b).norm();
+
+		// should this be converted to a user-specified value?
```
[the remark this answers is not in the document]
Comment (maxpaik16) [reply src/polyfem/state/StateSolveLinear.cpp:100]: Sure

File: src/polyfem/utils/MatrixCache.hpp (hunk 2 above)
Comment (maxpaik16) [reply src/polyfem/utils/MatrixCache.hpp:91]: I see... I will remove the comment.

Comment (maxpaik16) [reply src/polyfem/utils/MatrixCache.hpp]: Is there any extra reference/documentation for the MatrixCache class? Are the other comments in this class accurate? I was having trouble understanding the different mappings/caches and how they are connected to each other.

**Pull request merged**